engine/runtime/src/main/java/org/enso/interpreter/node/expression/atom/GetFieldNode.java

```diff
@@ -4,16 +4,20 @@
 import com.oracle.truffle.api.frame.VirtualFrame;
 import com.oracle.truffle.api.nodes.NodeInfo;
 import com.oracle.truffle.api.nodes.RootNode;
+import org.enso.interpreter.EnsoLanguage;
 import org.enso.interpreter.runtime.callable.atom.Atom;
 import org.enso.interpreter.runtime.callable.function.Function;
 import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.callable.atom.StructsLibrary;
 
 @NodeInfo(shortName = "get_field", description = "A base for auto-generated Atom getters.")
 public class GetFieldNode extends RootNode {
   private final int index;
   private final String name;
   private final Type type;
 
+  private @Child StructsLibrary structs = StructsLibrary.getFactory().createDispatched(10);
```

Comment:
We don't, but we don't have benchmarks that saturate this at all. Intuitively: I chose 10 because it is enough to handle all 2-field cases and it is much cheaper to go through the cache checks here 10 times than fire the uncached version even once.